plugin/storage/badger/factory.go

```diff
@@ -150,7 +161,8 @@ func (f *Factory) CreateDependencyReader() (dependencystore.Reader, error) {
 
 // Close Implements io.Closer and closes the underlying storage
 func (f *Factory) Close() error {
-	f.maintenanceDone <- true
+	f.maintenanceDone <- true // maintenance close
+	f.maintenanceDone <- true // metrics close
```

Comment:
Shouldn't it be `metricsTicker`?

Reply:
No, the ticker is closed when the select ends.

Reply:
why are we writing twice to the same channel?  If it's only meant to be read-once for exit signal, then I think the usual practice for that is to close the channel.